evadb/optimizer/optimizer_utils.py

```diff
@@ -280,12 +310,16 @@ def enable_cache_on_expression_tree(
 
 
 def check_expr_validity_for_cache(expr: FunctionExpression):
-    return (
-        expr.name in CACHEABLE_FUNCTIONS
-        and not expr.has_cache()
-        and len(expr.children) <= 1
-        and isinstance(expr.children[0], TupleValueExpression)
-    )
+    valid = expr.name in CACHEABLE_FUNCTIONS and not expr.has_cache()
+    if len(expr.children) == 1:
+        # Normal function that only takes one parameter.
+        valid &= isinstance(expr.children[0], TupleValueExpression)
+    elif len(expr.children) == 2:
+        # LLM-based function that takes two parameters.
+        valid &= isinstance(expr.children[0], ConstantValueExpression) and isinstance(
+            expr.children[1], TupleValueExpression
```

Comment:
Is there a particular reason we need to enforce the number of keys and type here? Do `FUNC(constant, constant)` or `Func(tuple, tuple)` not work with cache now?

Reply:
I think we don't have functions that act like this for now. We can get rid of the check now, but probably need to create more test cases to make sure it works. Or we can get rid of it later once we do have concrete functions to use. Thoughts?

Reply:
Sounds good.